### DISCUSSION TITLE
Pantry requests slower then 10 seconds

### DISCUSSION THREAD

**Discussion by davestaab** [imRohan/Pantry#189] [General]
I've got a small pantry store (3 buckets) with a small amount of data. It's taking more then 10 seconds to respond to requests which is causing problems with other services. 

I'm using [netlify ](https://www.netlify.com/) functions and they timeout after 10 seconds. Since pantry is currently taking longer then that, I'm kind of dead in the water. It wasn't taking this long a few weeks ago. 

Any idea if this situation can be improved? 

Thanks so much for this service! It hits a a sweet spot of low friction data storage for small projects. It's a great solution! Keep up the good work.

**Comment by imRohan** [comment]
Hi @davestaab 

Unfortunately, Pantry is now experiencing bottlenecks due to the infrastructure it's running on.

Many users have begin rapidly polling their Pantry's causing a large influx of requests to come in, which my AWS instance just can not handle. Due to this I've had to implement strict rate limiting rules (which you've experienced, sorry) to prevent the service from going down.

**Reply by davestaab** [reply]
I'm sorry to hear this. 😞 

Thanks for the reply.

**Comment by imRohan** [comment]
This is now fixed.
